### PR TITLE
PORT-7715 | Make Dora Metrics Script Platform Agnostic

### PR DESCRIPTION
**Pull request by mk-armah**
# Description

The change presents a improvement to the existing implementation by making the scripts for computing dora metrics independent of github.

## Updated docs pages
- Resource Path - (/build-your-software-catalog/custom-integration/api/ci-cd/github-workflow/guides/dora-metrics)

### CODE REVIEW

File: docs/build-your-software-catalog/custom-integration/api/ci-cd/github-workflow/guides/dora-metrics.md [hunk 1]
```diff
@@ -223,31 +223,17 @@ jobs:
...
 
       - name: Compute PR Metrics
-        env:
-          GITHUB_TOKEN: ${{ secrets.PATTOKEN }}
-          REPOSITORY: ${{ matrix.repository }}
-          OWNER: ${{ matrix.owner }}
         run: |
-          python dora/calculate_pr_metrics.py
-
+          python dora/calculate_pr_metrics.py  --owner ${{ matrix.owner }} --repo ${{ matrix.repository }} --token ${{ secrets.PATTOKEN }} --timeframe ${{env.TIMEFRAME_IN_DAYS}} --platform github-actions
```
Comment (Hardik-Murdia) [inline docs/build-your-software-catalog/custom-integration/api/ci-cd/github-workflow/guides/dora-metrics.md:227]: in compute PR metrics run script I think we need --ignore_workflows

File: docs/build-your-software-catalog/custom-integration/api/ci-cd/github-workflow/guides/dora-metrics.md [hunk 2]
```diff
@@ -630,66 +567,63 @@ class DeploymentFrequency:
...
-import asyncio
-import time
+import json
+from github import Github
+import argparse
+import logging
 
-PAGE_SIZE = 100
+#Throttling, set to None to restore default behavior
+SECONDS_BETWEEN_REQUESTS=0.12
```
Comment (Hardik-Murdia) [inline docs/build-your-software-catalog/custom-integration/api/ci-cd/github-workflow/guides/dora-metrics.md:623]: what is the significance of this 0.12 and I didn't get the throttling comment here ?

File: docs/build-your-software-catalog/custom-integration/api/ci-cd/github-workflow/guides/dora-metrics.md [hunk 3]
```diff
@@ -911,21 +786,27 @@ class LeadTimeForChanges:
...
+    report = lead_time_for_changes()
+    print(report)
+    
+    if args.platform == "github-actions":
+       with open(os.getenv("GITHUB_ENV"), "a") as github_env:
+           github_env.write(f"lead_time_for_changes_report={report}\n")
 ```
 </details>
 
 Congrats 🎉 You've successfully scheduled a GitHub action to periodically ingest estimated `DORA Metrics` for GitHub repository(s).
```
Comment (Hardik-Murdia) [inline docs/build-your-software-catalog/custom-integration/api/ci-cd/github-workflow/guides/dora-metrics.md:812]: this note can be changes as now you have schedule a job to periodically ingest estimated DORA metics for git repos.

**Review by MPTG94 — APPROVED** [review]
LGTM

File: docs/build-your-software-catalog/custom-integration/api/ci-cd/github-workflow/guides/dora-metrics.md (hunk 2 above)
Comment (MPTG94) [inline docs/build-your-software-catalog/custom-integration/api/ci-cd/github-workflow/guides/dora-metrics.md:623]: Are these throttling parameters needed?

File: docs/build-your-software-catalog/custom-integration/api/ci-cd/github-workflow/guides/dora-metrics.md [hunk 4]
```diff
@@ -507,103 +498,49 @@ if __name__ == "__main__":
...
-import httpx
-from loguru import logger
-import asyncio
+from github import Github
+import argparse
+import logging
 
-PAGE_SIZE = 100
+#Throttling, set to None to restore default behavior
+SECONDS_BETWEEN_REQUESTS=0.12
```
Comment (MPTG94) [inline docs/build-your-software-catalog/custom-integration/api/ci-cd/github-workflow/guides/dora-metrics.md:507]: Are these throttling parameters needed?

File: docs/build-your-software-catalog/custom-integration/api/ci-cd/github-workflow/guides/dora-metrics.md (hunk 3 above)
Comment (MPTG94) [inline docs/build-your-software-catalog/custom-integration/api/ci-cd/github-workflow/guides/dora-metrics.md:812]:
```suggestion
Congrats 🎉 You've successfully created a scheduled GitHub workflow that periodically calculates and ingests `DORA Metrics` for GitHub repository(s).

```

**Review by MPTG94 — CHANGES_REQUESTED** [review]
Please take a look at my comments

File: docs/build-your-software-catalog/custom-integration/api/ci-cd/github-workflow/guides/dora-metrics.md (hunk 2 above)
Comment (mk-armah) [reply docs/build-your-software-catalog/custom-integration/api/ci-cd/github-workflow/guides/dora-metrics.md:623]: Not neccessarily, but the default values are a bit high so i overwrote them with lower values

File: docs/build-your-software-catalog/custom-integration/api/ci-cd/github-workflow/guides/dora-metrics.md (hunk 1 above)
Comment (mk-armah) [reply docs/build-your-software-catalog/custom-integration/api/ci-cd/github-workflow/guides/dora-metrics.md:227]: It was deliberately left out in the github action to enforce running with workflows by default

Comment (mk-armah) [reply docs/build-your-software-catalog/custom-integration/api/ci-cd/github-workflow/guides/dora-metrics.md:227]: to run the without computing workflow, we can set that parameter

File: docs/build-your-software-catalog/custom-integration/api/ci-cd/github-workflow/guides/dora-metrics.md (hunk 2 above)
Comment (mk-armah) [reply docs/build-your-software-catalog/custom-integration/api/ci-cd/github-workflow/guides/dora-metrics.md:623]: https://pygithub.readthedocs.io/en/stable/changes.html#version-2-1-0-september-29-2023:~:text=%C2%B6-,Request%20throttling,-This%20release%20introduces

**Review by MPTG94 — APPROVED** [review]
LGTM

**Pull request merged**